scripts/version-utils.js

```diff
@@ -9,28 +9,147 @@
 
 const VERSION_REGEX = /^v?((\d+)\.(\d+)\.(\d+)(?:-(.+))?)$/;
 
-function parseVersion(versionStr) {
-  const match = versionStr.match(VERSION_REGEX);
-  if (!match) {
-    throw new Error(
-      `You must pass a correctly formatted version; couldn't parse ${versionStr}`,
-    );
-  }
+/**
+ * Parses a version string and performs some checks to verify its validity.
+ * A valid version is in the format vX.Y.Z[-KKK] where X, Y, Z are numbers and KKK can be something else.
+ * The `builtType` is used to enforce that the major version can assume only specific
+ * values.
+ *
+ * Parameters:
+ * - @versionStr the string representing a version
+ * - @buildType the build type. It can be of values: `dry-run`, `release`, `nightly`
+ *
+ * Returns: an object with the shape:
+ * ```
+ * {
+ *   version: string,
+ *   major: number,
+ *   minor: number,
+ *   patch: number,
+ *   prerelease: string
+ * }
+ * ```
+ */
```

Comment:
I think we should also have a comment or a link to a reference somewhere that clearly shows an example of of version for all types, such as
- stable `0.68.1`
- stable prerelease `0.70.0-rc.0`
- nightly `0.0.0-20221116-2018-0bc4547fc`
- dryrun `1000.0.0`
- etc (if we have other cases)